src/app/components/Curation/HierarchicalGrid/index.tsx

```diff
@@ -43,9 +46,30 @@ const HiearchicalGrid = ({
   const videoTranslation = path(['media', 'video'], translations);
   const photoGalleryTranslation = path(['media', 'photogallery'], translations);
   const durationTranslation = path(['media', 'duration'], translations);
+
   if (!summaries || summaries.length < 3) return null;
 
   const promoItems = summaries.slice(0, 12);
+
+  const buildPromoEventTrackingData = (promo: Summary, i: number) => {
+    const itemTracker = {
+      type: 'hierarchical-curation-grid-promo',
+      text: promo.title,
+      position: i + 1,
+      resourceId: promo.id,
+      ...(promo.type && { mediaType: promo.type }),
+      ...(promo.duration && {
+        duration: moment.duration(promo.duration, 'seconds').asMilliseconds(),
+      }),
+    };
+    return {
+      itemTracker,
+      ...eventTrackingData,
+    };
+  };
+
+  const getClickTrackerHandler = useClickTrackerHandler;
```

Comment:
This is interesting, does it work and pass different data for each of the promos? Haven't seen it used like this, but cool to know it works for in this type of situation.